src/Profiles.ts

```diff
@@ -446,141 +382,30 @@ export class Profiles {
         }
     }
 
-    public async editSession(profileLoaded: IProfileLoaded, profileName: string): Promise<any| undefined> {
-        const editSession = profileLoaded.profile;
-        const editURL = editSession.host+ ":" + editSession.port;
-        const editUser = editSession.user;
-        const editPass = editSession.password;
-        const editrej = editSession.rejectUnauthorized;
-        let updUser: string;
-        let updPass: string;
-        let updRU: boolean;
-        let updPort: number;
-        let updUrl: any;
-
-        const schema: {} = await this.getSchema(profileLoaded.type);
-        const schemaArray = Object.keys(schema);
-
-        const updSchemaValues: any = {};
+    public async editSession(profileLoaded: IProfileLoaded, profileName: string): Promise<IProfile | void> {
+        const schema = await this.getSchema("zosmf");
+        const updSchemaValues = await ZoweExplorerApiRegister.getCommonApi(profileLoaded)
+                                                             .collectProfileDetails(null,
+                                                                                    profileLoaded.profile,
+                                                                                    schema);
```

Comment:
This is another new piece to the added to the getCommanApi that extenders won't have yet and is a breaking change for editProfile and createProfile

Reply:
Not really sure what to do about this...the prompting function only exists inside collectProfileDetails now. And I don't really want to put it back into createNewConnection, because then the prompting logic is duplicated in 2 places...

Reply:
I don't understand why you changed the creation and editing of profiles to rely on code provided by an extender.  anything in the ZoweExplorerZosmfApi.ts file will not be read by the ftp or other extentions
Extensions have to have their own ...Api.ts file in their extension to handle things, but creation and editing should be handled by ZE instead of the extensions

Reply:
Ok collectProfileDetails is moved back into profiles